### PR TITLE
Add more version information

### LINKED ISSUE #4175: [FR] Add Installer type, branch and publisher to version info
Might be a checkpoint for system checks - depending on no installer type being detected.

### PR DESCRIPTION
**Pull request by matmair**
This PR adds more version information to packages:
- installer
- branch
- platform

and makes commit hash and date information available for packaged apps.

Closes #4175

### CODE REVIEW

File: InvenTree/InvenTree/version.py [hunk 1]
```diff
@@ -130,3 +130,39 @@ def inventreeCommitDate():
 
     commit_dt = dt.fromtimestamp(main_commit.commit_time) + td(seconds=main_commit.commit_timezone)
     return str(commit_dt.date())
+
+
+def inventreeInstaller():
+    """Returns the installer for the running codebase - if set."""
+    # First look in the environment variables, e.g. if running in docker
+
+    installer = os.environ.get('INVENTREE_PKG_INSTALLER', '')
```
Comment (wolflu05) [inline InvenTree/InvenTree/version.py:139]: Why aren't you using None as default and return that, it's mich shorter?

Comment (matmair) [reply InvenTree/InvenTree/version.py:139]: Because the PR is not done yet and there is more to come

**Review by SchrodingersGat — APPROVED** [review]

**Comment by matmair** [comment]
@SchrodingersGat please wait for one more commit with merging - I have to extend the docs so I do not forget what which packager code is

**Comment by matmair** [comment]
New sample output:
```
# Version Information:
InvenTree-Version: 0.12.0 dev
Django Version: 3.2.19
Commit Hash: 0a788b6a
Commit Date: 2023-06-11
Database: sqlite3
Debug-Mode: False
Deployed using Docker: False
Platform: Linux-5.15.0-67-generic-x86_64-with-glibc2.35
Installer: PKG
Target: ubuntu:20.04
Active plugins: [{'name': 'InvenTreeBarcode', 'slug': 'inventreebarcode', 'version': '2.0.0'}, {'name': 'InvenTreeCoreNotificationsPlugin', 'slug': 'inventreecorenotificationsplugin', 'version': '1.0.0'}]
```

**Comment by matmair** [comment]
@inventree/triage ready for review and merge

**Review by SchrodingersGat — APPROVED** [review]

**Pull request merged**